### ISSUE TITLE
Change the image formats to nextgen formats, i.e: WebP and AVIF (reference). Browser compatibility might be an issue so in that case, maybe we can reduce/compress the image size to 30-150 KB at max.

### ISSUE THREAD

**Issue by pdpinch** [mitodl/ocw-hugo-themes#636]

**Comment by pdpinch** [comment]
I'd like to narrow the scope of this just to the CHP page.

**Comment by pdpinch** [comment]
Can we use Hugo to process the images? See https://gohugo.io/content-management/image-processing/#target-format

**Comment by MAbdurrehman12** [comment]
@pdpinch Looked into it and it seems like there are a few things to discuss beforehand. I think I'll get in touch with @gumaerc today or maybe we can have a checkin later on

**Comment by gumaerc** [comment]
@pdpinch @MAbdurrehman12 I just read over this ticket and I have a few thoughts.  First of all, the images in `www` could be easily batch converted using GIMP or Photoshop, which I can do easily enough and put into a PR. As far as processing images using Hugo, I don't think we'd be able to take advantage of that aside from images directly embedded into a Hugo build.  Since our images are stored in S3 and a link to the image is simply written in during the build process, there's nothing there for Hugo to process.  What we would likely have to do (in addition to having `ocw-studio` convert new images) is download all the existing images in S3, convert and re-upload them, then change the references and remove the old images.  Another option would be to use an [S3 image processing lambda](https://aws.amazon.com/blogs/compute/resize-images-on-the-fly-with-amazon-s3-aws-lambda-and-amazon-api-gateway/) to convert images on the fly that are requested from the production bucket where the site is deployed.  This would allow us to essentially change nothing about how `ocw-studio` handles the images aside from having to change the references to the images to point to `.webp` instead of `.jpg` or whatever format they were uploaded in.

**Comment by MAbdurrehman12** [comment]
> the images in www could be easily batch converted using GIMP or Photoshop

Yeah, I also found an online tool through which we can easily convert in webp and reduce their size, in batch.

> Since our images are stored in S3 and a link to the image is simply written in during the build process, there's nothing there for Hugo to process.

This was exactly what I had in mind but just wanted to reconfirm

and I agree with your other thoughts stated above. Resizing and converting images on the fly using lamda seems to be a good approach as it has a couple of advantages, I was wondering whether this approach will have any con(s) or any impact on the performance?

**Comment by pdpinch** [comment]
I don't think I understand [S3 image processing lambda](https://aws.amazon.com/blogs/compute/resize-images-on-the-fly-with-amazon-s3-aws-lambda-and-amazon-api-gateway/) option. When would that be invoked? 

I'm trying to keep in mind our alternate distributions schemes -- downloading courses zips, mirror drives, etc -- and I'd prefer an approach that allows us to .webp (or whatever) consistently across all of these.

**Comment by gumaerc** [comment]
@pdpinch You raise a good point about the alternate delivery methods here.  From what I understand about lambda, there are a number of ways you can implement it.  The first example they give show a user requesting an object from a bucket that doesn't exist, which triggers a call to a resize API that finds the image they were looking for and resizes it on the fly.  For example, a user requests `image_sm.jpg` and only `image.jpg` exists on the server, so a call is made to the resize API automatically on the fly and the user is delivered the image.

In our case, I think we'd need to implement lambda functionality on the `ocw-studio` buckets themselves, not the live site buckets.  In theory, we could allow users to upload images in whatever format they want, then we could set it up so that when Concourse goes to build the site any time it requests any kind of image, it's delivered `.webp` images instead.  There are other things we can potentially use lambda for, like delivering the course zips themselves on the fly as a user requests them.  The lambda function could do whatever it needs to do to package the course in a ZIP, whatever we eventually decide we should be putting in there, and then deliver it to the user.

This is all hypothetical of course, and I don't have a whole lot of practical experience using lambda.  I would need access to the AMS lambda console to play with that and see what's actually possible.  It might end up being that it's not the best way to go and we should just convert the assets upon upload in `ocw-studio`.  It seems like lambda is the best option if you need to do stuff "on demand" though so we should keep it in mind for those kinds of use cases.
